src/controller/static_mates.controller.ts

```diff
@@ -1,72 +1,114 @@
+import { UpdatedAt } from 'sequelize-typescript';
 import { StaticMate } from '../models/static_mates.model.js'
 
 
-// /////Read or Pull
+///////Read or Pull all
 export const findAllStaticMates = async (req, res) => {
     try {
         console.log("Calling find all for Static Mates")
-
         const allStaticMates:StaticMate[] = await StaticMate.findAll();
-
         res.status(200).json({message: allStaticMates})
     } catch (error) {
         res.status(500).json({ message: error.message });
     }
 }
 
-// ///create
-// export const getSingle_Static_Mate = async (res, req) => {
-//     try {
-//         const { id } = req.params;
-//         const single_static_mate = await Static_Mates.findById(id);
-//         res.status(200).json(single_static_mate);
-//     } catch (error) {
-//         res.status(500).json({ message: error.message });
-//     }
-// }
-// export const createStatic_Mate = async (req, res) => {
-//     try {
-//         const single_static_mate = await Static_Mates.create(req.body);
-//         res.status(200).json(single_static_mate);
-//     } catch (error) {
-//         res.status(500).json({ message: error.message })
-//     }
-// };
-
+//Pull a SINGLE PERSON NEEDS TO BE FIXED
+export const findSingleStaticMate = async (req, res) => {
+    try {
+        console.log("Calling find all for Static Mates")
+        const allStaticMates:StaticMate[] = await StaticMate.findAll();
+        res.status(200).json({message: allStaticMates})
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+}
 
-// ///Update
-// export const updateStatic_Mate = async (req, res) => {
-//     try {
-//         const { id } = req.params;
+///create
 
-//         const single_static_mate = await Static_Mates.findByIdAndUpdate(id, req.body);
+export const createSingleStaticMate = async (req, res) => {
+    try {
+        //properties on the query object and the query object is property on the request object 
+        const { static_name, raid_member_role } = req.body;
+        //console.log("req.body", req.body)
+        //find a single pesron
+        const sM = await StaticMate.findOne({ where: {name: static_name} });
+        if (sM){
+            throw Error("There was already a user with that name")
+            return
+        }
+        //this is the set up of paramaters for postman
+        const single_static_mate = await StaticMate.create({
```

Comment:
.

Reply:
Create as a whole updated with payload and updated 200 and 500 error messages